src/include/common/object_pool.h

```diff
@@ -63,15 +144,23 @@ class ObjectPool {
    * @param obj pointer to object to release
    */
   void Release(T *obj) {
-    if (reuse_queue_.UnsafeSize() > reuse_limit_)
+    SpinLatch::ScopedSpinLatch guard(&latch_);
+    if (reuse_queue_.size() >= reuse_limit_) {
       alloc_.Delete(obj);
-    else
-      reuse_queue_.Enqueue(std::move(obj));
+      current_size_--;
+    } else {
+      reuse_queue_.push(obj);
+    }
   }
 
  private:
   Allocator alloc_;
-  ConcurrentQueue<T *> reuse_queue_;
-  const uint64_t reuse_limit_;
+  SpinLatch latch_;
+  std::queue<T *> reuse_queue_;
```

Comment:
Add a TODO if we should explicitly pass a second template parameter to define the backing container for the std::queue. We could extend the ObjectPool tests to behave more like a benchmark and then measure each backing container.

Reply:
So you are saying that we don't need to use std::queue here and, instead, we can have a second template argument that defines the underlying container for holding reusable objects. 

If I understand correctly, I agree with you. We don't need to reuse objects in a FIFO pattern. All we need is a container that can store whatever we give to him and give something back when we ask for it. I'll add a TODO